server/base/src/test/java/org/apache/accumulo/server/constraints/MetadataConstraintsTest.java

```diff
@@ -381,6 +384,41 @@ public void testBulkFileCheck() {
     assertEquals(Short.valueOf((short) 8), violations.get(0));
     assertNotNull(mc.getViolationDescription(violations.get(0)));
 
+    // Bad Json - only path (old format) so should fail parsing
+    m = new Mutation(new Text("0;foo"));
+    m.put(BulkFileColumnFamily.NAME, new Text("hdfs://1.2.3.4/accumulo/tables/2a/t-0003/someFile"),
+        new Value("5"));
+    violations = mc.check(createEnv(), m);
+    assertNotNull(violations);
+    assertEquals(1, violations.size());
+    assertEquals(Short.valueOf((short) 9), violations.get(0));
+    assertNotNull(mc.getViolationDescription(violations.get(0)));
+
```

Comment:
I was experimenting with these test and replacing startRow with empty string produces json that looks like the following.

```json
{"path":"hdfs://1.2.3.4/accumulo/tables/2a/t-0003/someFile","":"","endRow":""}
```

In the json above the startRow field is missing and an unexpected field with the empty string for a name exists. Thinking it would be nice to add another test case where startRow is missing and there is not an unexpected field.  Did that below, did not attempt a regex to create the json though.

```suggestion
    // Bad Json - test startRow key missing so validation should fail
    m = new Mutation(new Text("0;foo"));
    m.put(BulkFileColumnFamily.NAME, new Text("{\"path\":\"hdfs://1.2.3.4/accumulo/tables/2a/t-0003/someFile\",\"endRow\":\"\"}"),
            new Value("5"));
    violations = mc.check(createEnv(), m);
    assertNotNull(violations);
    assertEquals(1, violations.size());
    assertEquals(Short.valueOf((short) 9), violations.get(0));
    assertNotNull(mc.getViolationDescription(violations.get(0)));
```

Looking into this I was wondering if empty string is a valid field name in json, found discussion about this and it appears it is.

Reply:
Good idea, I will work on these and have an update shortly.